ext/json/json.c

```diff
@@ -270,6 +292,66 @@ PHP_FUNCTION(json_decode)
 }
 /* }}} */
 
+/* {{{ Validates if an string contains a valid json */
+PHP_FUNCTION(json_validate)
+{
+	char *str;
+	size_t str_len;
+	zend_long depth = PHP_JSON_PARSER_DEFAULT_DEPTH;
+	zend_long options = 0;
+
+	ZEND_PARSE_PARAMETERS_START(1, 3)
+		Z_PARAM_STRING(str, str_len)
+		Z_PARAM_OPTIONAL
+		Z_PARAM_LONG(depth)
+		Z_PARAM_LONG(options)
+	ZEND_PARSE_PARAMETERS_END();
+
+
+	if (options != 0) {
+		zend_long tmp_options = options;
+		tmp_options ^= (PHP_JSON_INVALID_UTF8_IGNORE | PHP_JSON_THROW_ON_ERROR);
+
+		if (tmp_options != 0) {
+			if (!((tmp_options == PHP_JSON_INVALID_UTF8_IGNORE) || (tmp_options == PHP_JSON_THROW_ON_ERROR))) {
```

Comment:
```suggestion
		if (options & ~(PHP_JSON_INVALID_UTF8_IGNORE | PHP_JSON_THROW_ON_ERROR))
```

Reply:
Done, and adjusted to the requirement to not throw exception, so now only valid flag is PHP_JSON_INVALID_UTF8_IGNORE

But trust that I have learn something out of your code review. Thanks.